# MIT License
#
# Copyright (c) 2025 Advanced Micro Devices, Inc.
#
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in all
# copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.
diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
index 3c2d3f6..5271281 100644
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -274,6 +274,10 @@ targets:
     ]=]
 )

+if(TARGET conda_env)
+  install(TARGETS conda_env EXPORT ucxx-exports)
+endif()
+
 rapids_export(
   INSTALL ucxx
   EXPORT_SET ucxx-exports ${_components_export_string}
diff --git a/cpp/src/internal/request_am.cpp b/cpp/src/internal/request_am.cpp
index d45dee6..b2b3750 100644
--- a/cpp/src/internal/request_am.cpp
+++ b/cpp/src/internal/request_am.cpp
@@ -22,7 +22,7 @@ RecvAmMessage::RecvAmMessage(internal::AmData* amData,
   : _amData(amData), _ep(ep), _request(request)
 {
   std::visit(data::dispatch{
-               [this, buffer](data::AmReceive& amReceive) { amReceive._buffer = buffer; },
+               [buffer](data::AmReceive& amReceive) { amReceive._buffer = buffer; },
                [](auto) { throw std::runtime_error("Unreachable"); },
              },
              _request->_requestData);
diff --git a/cpp/src/request_am.cpp b/cpp/src/request_am.cpp
index 6444989..8d19e78 100644
--- a/cpp/src/request_am.cpp
+++ b/cpp/src/request_am.cpp
@@ -457,7 +457,7 @@ void RequestAm::populateDelayedSubmission()
   };

   std::visit(data::dispatch{
-               [this, &log](data::AmSend amSend) {
+               [&log](data::AmSend amSend) {
                  log(amSend._buffer, amSend._length, amSend._memoryType);
                },
                [](auto) { throw std::runtime_error("Unreachable"); },
diff --git a/cpp/src/request_mem.cpp b/cpp/src/request_mem.cpp
index 74817f9..f1846d0 100644
--- a/cpp/src/request_mem.cpp
+++ b/cpp/src/request_mem.cpp
@@ -180,10 +180,10 @@ void RequestMem::populateDelayedSubmission()
     };

   std::visit(data::dispatch{
-               [this, &log](data::MemPut memPut) {
+               [&log](data::MemPut memPut) {
                  log(memPut._buffer, memPut._length, memPut._remoteAddr, memPut._rkey);
                },
-               [this, &log](data::MemGet memGet) {
+               [&log](data::MemGet memGet) {
                  log(memGet._buffer, memGet._length, memGet._remoteAddr, memGet._rkey);
                },
                [](auto) { throw std::runtime_error("Unreachable"); },
diff --git a/cpp/src/request_stream.cpp b/cpp/src/request_stream.cpp
index 704ab8a..46555d7 100644
--- a/cpp/src/request_stream.cpp
+++ b/cpp/src/request_stream.cpp
@@ -144,8 +144,8 @@ void RequestStream::populateDelayedSubmission()

   std::visit(
     data::dispatch{
-      [this, &log](data::StreamSend streamSend) { log(streamSend._buffer, streamSend._length); },
-      [this, &log](data::StreamReceive streamReceive) {
+      [&log](data::StreamSend streamSend) { log(streamSend._buffer, streamSend._length); },
+      [&log](data::StreamReceive streamReceive) {
         log(streamReceive._buffer, streamReceive._length);
       },
       [](auto) { throw std::runtime_error("Unreachable"); },
diff --git a/cpp/src/request_tag.cpp b/cpp/src/request_tag.cpp
index 9655c44..784ab30 100644
--- a/cpp/src/request_tag.cpp
+++ b/cpp/src/request_tag.cpp
@@ -194,10 +194,10 @@ void RequestTag::populateDelayedSubmission()
   };

   std::visit(data::dispatch{
-               [this, &log](data::TagSend tagSend) {
+               [&log](data::TagSend tagSend) {
                  log(tagSend._buffer, tagSend._length, tagSend._tag, TagMaskFull);
                },
-               [this, &log](data::TagReceive tagReceive) {
+               [&log](data::TagReceive tagReceive) {
                  log(tagReceive._buffer, tagReceive._length, tagReceive._tag, tagReceive._tagMask);
                },
                [](auto) { throw std::runtime_error("Unreachable"); },
diff --git a/cpp/src/worker.cpp b/cpp/src/worker.cpp
index d365737..7998065 100644
--- a/cpp/src/worker.cpp
+++ b/cpp/src/worker.cpp
@@ -517,7 +517,7 @@ size_t Worker::cancelInflightRequests(uint64_t period, uint64_t maxAttempts)
         continue;

       if (!registerGenericPost(
-            [this, &inflightRequestsToCancel, &cancelSuccess]() {
+            [&inflightRequestsToCancel, &cancelSuccess]() {
               cancelSuccess = inflightRequestsToCancel->getCancelingSize() == 0;
             },
             period))
